tooling/nargo_cli/src/cli/fuzz_cmd.rs

```diff
@@ -0,0 +1,489 @@
+        match &pattern {
+            FunctionNameMatch::Exact(pattern) => {
+                let single_pattern = pattern[0].clone();
+                return Err(CliError::Generic(format!(
+                    "Found 0 fuzzing_harnesses matching input '{single_pattern}'.",
+                )));
+            }
+            FunctionNameMatch::Contains(pattern) => {
+                let single_pattern = pattern[0].clone();
+                return Err(CliError::Generic(format!(
+                    "Found 0 fuzzing_harnesses containing '{single_pattern}'.",
+                )));
+            }
+            // If we are running all tests in a crate, having none is not an error
+            FunctionNameMatch::Anything => {}
+        };
+    }
+
+    if fuzzing_report.iter().any(|(_, status)| status.failed()) {
+        Err(CliError::Generic(String::new()))
+    } else {
+        Ok(())
+    }
+}
+
+fn list_harnesses(
+    file_manager: &FileManager,
+    parsed_files: &ParsedFiles,
+    package: &Package,
+    fn_name: &FunctionNameMatch,
+    compile_options: &CompileOptions,
+) -> Result<Vec<String>, CliError> {
+    let fuzzing_harnesses = get_fuzzing_harnesses_in_package(
+        file_manager,
+        parsed_files,
+        package,
+        fn_name,
+        compile_options,
+    )?;
+    Ok(fuzzing_harnesses)
+}
+
+#[allow(clippy::too_many_arguments)]
+fn run_fuzzers<S: BlackBoxFunctionSolver<FieldElement> + Default>(
+    file_manager: &FileManager,
+    parsed_files: &ParsedFiles,
+    package: &Package,
+    fn_name: &FunctionNameMatch,
+    show_output: bool,
+    foreign_call_resolver_url: Option<&str>,
+    root_path: Option<PathBuf>,
+    package_name: Option<String>,
+    compile_options: &CompileOptions,
+    fuzz_folder_config: &FuzzFolderConfig,
+    fuzz_execution_config: &FuzzExecutionConfig,
+) -> Result<Vec<(String, FuzzingRunStatus)>, CliError> {
+    let fuzzing_harnesses = get_fuzzing_harnesses_in_package(
+        file_manager,
+        parsed_files,
+        package,
+        fn_name,
+        compile_options,
+    )?;
+
+    let mut fuzzing_reports = Vec::new();
+    for fuzzing_harness_name in fuzzing_harnesses.into_iter() {
+        let status = run_fuzzing_harness::<S>(
+            file_manager,
+            parsed_files,
+            package,
+            &fuzzing_harness_name,
+            show_output,
+            foreign_call_resolver_url,
+            root_path.clone(),
+            package_name.clone(),
+            compile_options,
+            fuzz_folder_config,
+            fuzz_execution_config,
+        );
+        fuzzing_reports.push((fuzzing_harness_name, status));
+        // Display the latest report
+        display_fuzzing_report_and_store(
+            root_path.clone(),
+            fuzz_folder_config.fuzzing_failure_dir.clone(),
+            file_manager,
+            package,
+            compile_options,
+            &fuzzing_reports[(&fuzzing_reports.len() - 1)..fuzzing_reports.len()],
```

Comment:
It seems this will always be one element? If so, could `display_fuzzing_report_and_store` be changed to take just one of these?

Reply:
Done